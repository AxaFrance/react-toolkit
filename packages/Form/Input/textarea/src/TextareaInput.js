import React from 'react';
import PropTypes from 'prop-types';

import {
  Field,
  HelpMessage,
  FieldInput,
  FieldConstants as Constants,
  FormClassManager,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import Textarea from './Textarea';

const propTypes = {
  ...Constants.propTypes,
  value: PropTypes.string,
  rows: PropTypes.number,
  cols: PropTypes.number,
};

const defaultProps = {
  ...Constants.defaultProps,
  value: '',
  rows: 5,
  cols: 50,
};

const TextareaInput = props => {
  const {
    classModifier,
    message,
    children,
    helpMessage,
    id,
    disabled,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    messageType,
    isVisible,
    forceDisplayMessage,
    className,
    name,
    rows,
    cols,
    value,
    readOnly,
    placeholder,
    maxLength,
    minLength,
    ...otherProps
  } = props;
  const inputClassModifier = FormClassManager.getInputClassModifier(
    classModifier,
    children
  );
  const inputFieldClassModifier = FormClassManager.getFieldInputClassModifier(
    classModifier,
    disabled
  );
  const onChange = e => {
    props.onChange(e);
  };
  const rowModifier = `${classModifier} label-top`;
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={rowModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        helpMessage={helpMessage}
        message={message}
        messageType={messageType}
        className="af-form__textarea"
        classModifier={inputFieldClassModifier}>
        <Textarea
          name={name}
          id={inputId}
          rows={rows}
          cols={cols}
          onChange={onChange}
          value={value}
          readOnly={readOnly}
          disabled={disabled}
          placeholder={placeholder}
          classModifier={inputClassModifier}
          maxLength={maxLength}
          minLength={minLength}
          {...otherProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

TextareaInput.propTypes = propTypes;
TextareaInput.defaultProps = defaultProps;

export default TextareaInput;
