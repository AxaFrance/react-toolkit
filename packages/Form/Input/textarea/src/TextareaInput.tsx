import React, { ComponentProps, ReactNode } from 'react';
import {
  Field,
  HelpMessage,
  FieldInput,
  withInputClassModifier,
  InputManager,
} from '@axa-fr/react-toolkit-form-core';

import Textarea from './Textarea';

type Props = ComponentProps<typeof Field> &
  ComponentProps<typeof Textarea> & {
    inputFieldClassModifier: string;
    inputClassModifier: string;
    helpMessage?: string;
    children?: ReactNode;
  };

const TextareaInput = (props: Props) => {
  const {
    classModifier,
    message,
    children,
    helpMessage,
    id,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    messageType,
    isVisible,
    forceDisplayMessage,
    className,
    rows = 5,
    cols = 50,
    inputFieldClassModifier,
    inputClassModifier,
    ...textareaProps
  } = props;

  const rowModifier = `${classModifier} label-top`;
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={rowModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        className="af-form__textarea"
        classModifier={inputFieldClassModifier}>
        <Textarea
          id={inputId}
          rows={rows}
          cols={cols}
          classModifier={inputClassModifier}
          {...textareaProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

const enhance = withInputClassModifier(TextareaInput);
export default enhance;
