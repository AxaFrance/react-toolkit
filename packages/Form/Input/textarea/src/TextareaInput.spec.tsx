import React from 'react';
import renderer from 'react-test-renderer';
import TextareaInput from './TextareaInput';

describe('<SwitchInput>', () => {
  it('renders SwitchInput correctly', () => {
    const tree = renderer
      .create(
        <TextareaInput
          label="Image *"
          name="sample"
          id="iddateinput"
          value="ici le text"
          onChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
