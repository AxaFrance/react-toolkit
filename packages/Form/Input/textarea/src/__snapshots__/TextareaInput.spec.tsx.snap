// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchInput> renders SwitchInput correctly 1`] = `
<div
  className="row af-form__group af-form__group--undefined af-form__group--label-top"
>
  <div
    className="col-md-2"
  >
    <label
      className="af-form__group-label"
      htmlFor="iddateinput"
    >
      Image *
    </label>
  </div>
  <div
    className="col-md-10"
  >
    <div
      className="af-form__textarea af-form__textarea--undefined"
    >
      <textarea
        className="af-form__input-textarea af-form__input-textarea--undefined"
        cols={50}
        id="iddateinput"
        name="sample"
        onChange={[Function]}
        rows={5}
        value="ici le text"
      />
    </div>
    <small
      className="af-form__help"
    />
  </div>
</div>
`;
